src/Backend/Modules/Location/Command/CopyLocationWidgetsToOtherLocaleHandler.php

```diff
@@ -39,19 +34,26 @@ public function handle(CopyLocationWidgetsToOtherLocale $copyLocationWidgetsToOt
 
             // Replace the language of our widget
             $data['language'] = $copyLocationWidgetsToOtherLocale->toLocale->getLocale();
-            $currentWidget['data'] = serialize($data);
 
             // Save the old ID
-            $oldId = $currentWidget['id'];
-
-            // Unset the ID so we get a new one
-            unset($currentWidget['id']);
+            $oldId = $currentWidget->getId();
+
+            $newWidget = new ModuleExtra(
```

Comment:
wouldn't it be better to implement the clone functionality?
I think doctrine even automatically removes the id when cloning an entity

Reply:
In this case I would be cloning the object and the calling `update` with the same parameters as I do now for the constructor. I'm not immediately seeing the benefit of cloning is this case.

Reply:
I had missed the part where the value of data is potentially changed